### DISCUSSION TITLE
SCons and Python code not using the .py extension

### DISCUSSION THREAD

**Discussion by wahn** [zed-industries/zed#6497] [Feature Requests]
Hi,
before I ask my question I would like to give you some background on what I try to do.
I'm interested in The Language Server Protocol (LSP) and Tree-Sitter:
https://www.janwalter.org/jekyll/lsp/2022/10/01/language-servers.html
And while investigating both I stumbled across Zed. To test what's there already
I checked out two languages to experiment with, Python and C (or C++).
As a starting point I wanted to see how good I can explore existing Python code
and to make the connection between C and Python I thought I start with SCons:
https://scons.org
A Python based tool to compile (e.g. C code).

Now to my question. SCons uses Python files, which do not use the .py extension,
like e.g. _SConstruct_ and I would like to know how I can make Zed aware of that,
so I can still use all the other methods provided to explore Python code, e.g.
jump to a definition. It's a tricky one because those SConstruct files are called
from another file (in my case `/Users/jan/Library/Python/3.10/bin/scons`), 
which is also Python code and does not use the .py extension:

```python
#!/usr/local/opt/python@3.10/bin/python3.10
# -*- coding: utf-8 -*-
import re
import sys
from SCons.Script.Main import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```
In Emacs the syntax highlighting would be easy to do because you could simply open the file and do:

```
M-x python-mode
```

Maybe followed by:

```
M-x tree-sitter-hl-mode
```

To activate the Tree-Sitter syntax highlighting ...

Anyway, I asked my question and everything I have seen so far regarding Zed looks really promising!

Thanks,
Jan

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
